core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2582,13 +2582,21 @@ public void resetFile(long fileId)
 
   @Override
   public void setAttribute(AlluxioURI path, SetAttributeOptions options)
-      throws FileDoesNotExistException, AccessControlException, InvalidPathException {
+      throws FileDoesNotExistException, AccessControlException, InvalidPathException,
+      IOException {
     Metrics.SET_ATTRIBUTE_OPS.inc();
     // for chown
     boolean rootRequired = options.getOwner() != null;
     // for chgrp, chmod
     boolean ownerRequired =
         (options.getGroup() != null) || (options.getMode() != Constants.INVALID_MODE);
+    if (options.getOwner() != null && options.getGroup() != null) {
+      try {
+        checkUserBelongsToGroup(options.getOwner(), options.getGroup());
+      } catch (IOException e) {
+        throw new IOException("Could not setOwner for " + path.toString() + ".", e);
```

Comment:
When we send exceptions over the wire back to the client, we do not include the stack trace or the chain of causal exceptions. Therefore the error message itself must contain the root cause of the problem so that users can understand the failure. Here, the root cause is `e`, but by wrapping `e` in another `IOException`, the client will see nothing about `e`. To fix this, I suggest allowing `e` to propagate instead of catching it.